crates/common/mqtt_client/src/lib.rs

```diff
@@ -233,6 +257,24 @@ impl Client {
         })
     }
 
+    async fn pause_on_error(err: ConnectionError) {
```

Comment:
It would be good to use this function in the main loop of the `bg_process` from where this function has been extracted.